gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -293,4 +293,93 @@ fluid.registerNamespace("gpii.transformer");
         return (60 * (first.hours - second.hours) + first.minutes - second.minutes) < 0;
     };
 
+    fluid.defaults("gpii.transformer.jsonArrayToTaggedArray", {
+        gradeNames: "fluid.standardTransformFunction"
+    });
+
+    /**
+     * Creates a tagged array from a json array, using the tag specified in the 'transformSpec'.
+     * If the values in the array are present in the 'elems' array of the 'transformSpec',
+     * the tag that will be added for that element will be the value specified in
+     * 'transformSpec.hasElemVal.isTrue', otherwise the value will be 'transformSpec.hasElemVal.isFalse'.
+     *
+     * @param {Array} arr An array with the elements that will be transformed into the tagged array.
+     * @param {Object} transformSpec The required spec for the transform. It should have the form:
+     *  {
+     *     "tag": "tagValue",
+     *     "elems": [
+     *     ],
+     *     "hasElemVal": {
+     *         "isTrue": isTrueValue,
+     *         "isFalse": isFalseValue
+     *     }
+     *   }
+     * @return {Array} A tagged array result of processing the input array with the provided 'transformSpec'.
+     */
+    gpii.transformer.jsonArrayToTaggedArray = function (arr, transformSpec) {
+        var output = [];
+
+        // Check the transformSpec
+        gpii.transformer.jsonArrayToTaggedArray.checkSpec(transformSpec);
+
+        fluid.each(transformSpec.elems, function (elem) {
+            var arrayElem = {};
+
+            var hasElem = arr.indexOf(elem) !== -1;
+            fluid.set(arrayElem, "$t", elem);
```

Comment:
No need for fluid.set here, you just allocated it on line 326

Reply:
Simplified!